Sources/GRPCHTTP2TransportNIOPosix/GRPCHTTP2TransportNIOPosix.swift

```diff
@@ -15,7 +15,88 @@
  */
 
 import GRPCCore
+import GRPCHTTP2Core
+import NIOCore
+import NIOExtras
+import NIOPosix
 
-@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
-public struct GRPCHTTP2TransportNIOPosix {
+@available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
+public struct GRPCHTTP2TransportNIOPosix: ServerTransport {
+  private let configuration: GRPCHTTP2TransportNIOPosixConfiguration
+  private let eventLoopGroup: MultiThreadedEventLoopGroup
+  private let serverQuiescingHelper: ServerQuiescingHelper
+
+  init(configuration: GRPCHTTP2TransportNIOPosixConfiguration) {
+    self.configuration = configuration
+    self.eventLoopGroup = MultiThreadedEventLoopGroup(
+      numberOfThreads: self.configuration.maxConcurrentStreams
+    )
+    self.serverQuiescingHelper = ServerQuiescingHelper(group: self.eventLoopGroup)
+  }
+
+  public func listen(
+    _ streamHandler: @escaping (RPCStream<Inbound, Outbound>) async -> Void
+  ) async throws {
+    let serverChannel = try await ServerBootstrap(group: self.eventLoopGroup)
+      .serverChannelInitializer { channel in
+        let quiescingHandler = self.serverQuiescingHelper.makeServerChannelHandler(channel: channel)
+        return channel.pipeline.addHandler(quiescingHandler)
+      }
+      .bind(
+        host: self.configuration.host,
+        port: self.configuration.port,
```

Comment:
We should support more options than just host/port. The right API is probably `GRPCHTTP2Core.SocketAddress`